File: androidApp/src/main/kotlin/de/davidbattefeld/berlinskylarks/ui/scores/box/PitchingTable.kt
package de.davidbattefeld.berlinskylarks.ui.scores.box

import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.MaterialTheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import de.berlinskylarks.shared.data.model.PitchingMatchStats

@Composable
fun PitchingTable(teamName: String, matchStats: PitchingMatchStats) {
    val scrollState = rememberScrollState()
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 4.dp),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceContainer
        )
    ) {
        Column(
            modifier = Modifier
                .horizontalScroll(scrollState)
                .padding(8.dp)
        ) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                HeaderCell(text = "$teamName (Pitchers)", minWidth = 156.dp)
                HeaderCell(text = "IP", minWidth = 30.dp)
                HeaderCell(text = "BF", minWidth = 30.dp)
                HeaderCell(text = "AB", minWidth = 30.dp)
                HeaderCell(text = "H", minWidth = 30.dp)
                HeaderCell(text = "R", minWidth = 30.dp)
                HeaderCell(text = "ER", minWidth = 30.dp)
                HeaderCell(text = "K", minWidth = 30.dp)
                HeaderCell(text = "BB", minWidth = 30.dp)
                HeaderCell(text = "ERA", minWidth = 50.dp)
            }
            HorizontalDivider(modifier = Modifier.padding(vertical = 4.dp).width(440.dp))
            matchStats.lineup.forEach { player ->
                Row(verticalAlignment = Alignment.CenterVertically) {
                    val name = "${player.person.lastName}, ${player.person.firstName.firstOrNull() ?: ' '}."
                    val wls = player.values.winLossSave?.let { " ($it)" } ?: ""
                    BodyCell(text = name + wls, minWidth = 150.dp)
                    BodyCell(text = player.values.inningsPitched, minWidth = 30.dp)
                    BodyCell(text = player.values.battersFaced.toString(), minWidth = 30.dp)
                    BodyCell(text = player.values.atBats.toString(), minWidth = 30.dp)
                    BodyCell(text = player.values.hits.toString(), minWidth = 30.dp)
                    BodyCell(text = player.values.runs.toString(), minWidth = 30.dp)
                    BodyCell(text = player.values.earnedRuns.toString(), minWidth = 30.dp)
                    BodyCell(text = player.values.strikeouts.toString(), minWidth = 30.dp)
                    BodyCell(text = player.values.baseOnBallsAllowed.toString(), minWidth = 30.dp)
                    BodyCell(text = player.values.earnedRunsAverage ?: "", minWidth = 50.dp)
                }
            }
            HorizontalDivider(modifier = Modifier.padding(vertical = 4.dp).width(440.dp))
            Row(verticalAlignment = Alignment.CenterVertically) {
                BodyCell(text = "", minWidth = 150.dp)
                BodyCell(text = matchStats.sum.inningsPitched, minWidth = 30.dp)
                BodyCell(text = matchStats.sum.battersFaced.toString(), minWidth = 30.dp)
                BodyCell(text = matchStats.sum.atBats.toString(), minWidth = 30.dp)
                BodyCell(text = matchStats.sum.hits.toString(), minWidth = 30.dp)
                BodyCell(text = matchStats.sum.runs.toString(), minWidth = 30.dp)
                BodyCell(text = matchStats.sum.earnedRuns.toString(), minWidth = 30.dp)
                BodyCell(text = matchStats.sum.strikeouts.toString(), minWidth = 30.dp)
                BodyCell(text = matchStats.sum.baseOnBallsAllowed.toString(), minWidth = 30.dp)
                BodyCell(text = "", minWidth = 50.dp)
            }
        }
    }
}

File: iosApp/Berlin_Skylarks/Views/Scores/BoxScore/PitchingTable.swift
//
//  PitchingTable.swift
//  Berlin Skylarks
//
//  Created by David Battefeld on 25.09.25.
//

import Foundation
import SwiftUI
import sharedKit

struct PitchingTable: View {
    var teamName: String
    var matchStats: PitchingMatchStats

    var body: some View {
        ScrollView(.horizontal) {
            Grid(alignment: .center, horizontalSpacing: 4, verticalSpacing: 4) {
                // Header
                GridRow {
                    Text("\(teamName) (Pitchers)")
                        .frame(minWidth: 180, alignment: .leading)
                        .padding(.trailing, 6)
                    Text("IP").frame(minWidth: 56)
                    Text("BF").frame(minWidth: 56)
                    Text("AB").frame(minWidth: 56)
                    Text("H").frame(minWidth: 40)
                    Text("R").frame(minWidth: 40)
                    Text("ER").frame(minWidth: 56)
                    Text("K").frame(minWidth: 40)
                    Text("BB").frame(minWidth: 40)
                    Text("ERA").frame(minWidth: 64)
                }
                .font(.headline)

                Divider()
                // Lineup rows
                ForEach(Array(matchStats.lineup.enumerated()), id: \.offset) {
                    _,
                    player in
                    GridRow {
                        let firstInitial =
                            player.person.firstName.first.map { String($0) }
                            ?? ""
                        let name =
                            firstInitial.isEmpty
                            ? player.person.lastName
                            : "\(player.person.lastName), \(firstInitial)."
                        let wls = player.values.winLossSave ?? ""
                        Text(wls.isEmpty ? name : name + " (\(wls))")
                            .frame(minWidth: 180, alignment: .leading)
                        Text(player.values.inningsPitched).frame(minWidth: 56)
                        Text(String(player.values.battersFaced)).frame(
                            minWidth: 56
                        )
                        Text(String(player.values.atBats)).frame(minWidth: 56)
                        Text(String(player.values.hits)).frame(minWidth: 40)
                        Text(String(player.values.runs)).frame(minWidth: 40)
                        Text(String(player.values.earnedRuns)).frame(
                            minWidth: 56
                        )
                        Text(String(player.values.strikeouts)).frame(
                            minWidth: 40
                        )
                        Text(String(player.values.baseOnBallsAllowed)).frame(
                            minWidth: 40
                        )
                        Text(player.values.earnedRunsAverage ?? "").frame(
                            minWidth: 64
                        )
                    }
                }
                Divider()
                // Totals row
                GridRow {
                    Text("")
                        .frame(minWidth: 180, alignment: .leading)
                    Text(matchStats.sum.inningsPitched).frame(minWidth: 56)
                    Text(String(matchStats.sum.battersFaced)).frame(
                        minWidth: 56
                    )
                    Text(String(matchStats.sum.atBats)).frame(minWidth: 56)
                    Text(String(matchStats.sum.hits)).frame(minWidth: 40)
                    Text(String(matchStats.sum.runs)).frame(minWidth: 40)
                    Text(String(matchStats.sum.earnedRuns)).frame(minWidth: 56)
                    Text(String(matchStats.sum.strikeouts)).frame(minWidth: 40)
                    Text(String(matchStats.sum.baseOnBallsAllowed)).frame(
                        minWidth: 40
                    )
                    Text("").frame(minWidth: 64)
                }
            }
            .padding(.vertical, 4)
        }
    }
}
